cesium/internal/domain/index.go

```diff
@@ -78,20 +101,50 @@ func (idx *index) overlap(tr telem.TimeRange) bool {
 	return overlap
 }
 
-func (idx *index) update(ctx context.Context, p pointer) (err error) {
+func (idx *index) update(ctx context.Context, p pointer, persist bool) error {
 	_, span := idx.T.Bench(ctx, "update")
-	idx.mu.RLock()
+	idx.mu.Lock()
+
+	defer span.End()
+
 	if len(idx.mu.pointers) == 0 {
-		idx.mu.RUnlock()
-		return span.EndWith(RangeNotFound)
+		idx.mu.Unlock()
+		return span.Error(RangeNotFound)
 	}
 	lastI := len(idx.mu.pointers) - 1
 	updateAt := lastI
 	if p.Start != idx.mu.pointers[lastI].Start {
 		updateAt, _ = idx.unprotectedSearch(p.Start.SpanRange(0))
 	}
-	idx.mu.RUnlock()
-	return span.EndWith(idx.updateAt(ctx, updateAt, p))
+
+	ptrs := idx.mu.pointers
+	oldP := ptrs[updateAt]
+	if oldP.Start != p.Start {
+		idx.mu.Unlock()
+		return span.Error(RangeNotFound)
+	}
```

Comment:
Want to test when the condition on 122 is true as well? maybe the codecov

Reply:
Same as above